src/main/java/com/hrznstudio/titanium/component/IComponentHandler.java

```diff
@@ -0,0 +1,16 @@
+/*
+ * This file is part of Titanium
+ * Copyright (C) 2020, Horizon Studio <contact@hrznstudio.com>.
+ *
+ * This code is licensed under GNU Lesser General Public License v3.0, the full license text can be found in LICENSE.txt
+ */
+
+package com.hrznstudio.titanium.component;
+
+public interface IComponentHandler<T> {
+
+    void add(T component);
+
+    boolean accepts(Object component);
```

Comment:
What does accepts offer? It's impossible to properly cast to add just by knowing a Handler accepts something. It may be worthwhile to add a ```boolean add(Object component)``` or something like that? Depends on the usecase.